### PR TITLE
Disable intergrated security connections from c sharp strategies

### PR DESCRIPTION
**Pull request by gogo4ds**
Restrict access of the OJS Local Worker to SQL Server
Closes https://github.com/SoftUni-Internal/suls-issues/issues/3465

### CODE REVIEW

File: OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            {
+                sanitizedZipFile.Save(outputStream);
+
+                return outputStream.ToArray();
+            }
+        }
+
+        private static bool ExecutionContextContainsZipFile(ExecutionContext executionContext) =>
+            !string.IsNullOrWhiteSpace(executionContext.AllowedFileExtensions) &&
+              executionContext.AllowedFileExtensions.Contains(Constants.ZipFileExtension.Substring(1));
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs]: Indented with two extra whitespaces.

File: OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    using Ionic.Zip;
+
+    using OJS.Workers.Common;
+
+    public static class ExecutionContextExtensions
+    {
+        public static void SanitizeContent(this ExecutionContext executionContext)
+        {
+            var callingClassName = new StackFrame(1).GetMethod()?.DeclaringType?.Name;
```
Comment (jicata) [inline OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs]: Can't you instead use the [CallerMemberName attribute here](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.callermembernameattribute?view=netframework-4.7.2) ?

File: OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        private static void SanitizeDotNetCoreZipFile(ExecutionContext executionContext)
+        {
+            if (!ExecutionContextContainsZipFile(executionContext))
+            {
+                return;
+            }
+
+            const string connectionStringSearchPattern = @"(\.\s*UseSqlServer\s*\()(.*)(\))";
```
Comment (jicata) [inline OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs]: Extract into a constant elsewhere. Some global constant file or at least at the top of the declaring class

File: OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs [hunk 4]
```diff
@@ -12,11 +12,13 @@
...
-        public static void SanitizeContent(this ExecutionContext executionContext)
+        public static void SanitizeContent(
+            this ExecutionContext executionContext,
+            [CallerFilePath]string callerFilePath = null)
         {
-            var callingClassName = new StackFrame(1).GetMethod()?.DeclaringType?.Name;
-
-            switch (callingClassName)
+            var callerClassName = Path.GetFileNameWithoutExtension(callerFilePath);
+
```
Comment (achobanov) [inline OJS.Workers.ExecutionStrategies/Extensions/ExecutionContextExtensions.cs]: Whitespaces.

**Pull request merged**